NodeJS/src/AddressBookClient.ts

```diff
@@ -0,0 +1,46 @@
+import { SkillPriority } from './models'
+import { RestClient } from './RestClient'
+import { sleep } from './Utils'
+import { AddressBookResponse } from './models/AddressBookResponse'
+import { AddressBookSearchRequest } from './models/AddressBookSearchRequest'
+import { AddressBookSearchResponse } from './models/AddressBookSearchResponse'
+
+export class AddressBookClient {
+  private restClient: RestClient
+  private subAccountId: string
+
+  constructor(subAccountId: string, restClient: RestClient) {
+    this.restClient = restClient
+    this.subAccountId = subAccountId
+  }
+
+  /**
+   * @returns AddressBookResponse with all active contacts for a Sub-account
+   */
+  public async getAddressBook(): Promise<AddressBookResponse> {
+    return await this.restClient.getAddressBook()
+  }
+
+  /**
+   * @param AddressBookSearchRequest (optional) search by type, query, orderBy, orderDirection, page, pageSize
+   * @returns AddressBookSearchResponse with active contacts for a Sub-account according to search params
+   */
+  public async searchContacts(
+    request?: AddressBookSearchRequest
+  ): Promise<AddressBookSearchResponse> {
+    return await this.restClient.searchContacts(request)
+  }
+}
+
+async function createInstance(restClient: RestClient) {
+  const subAccountId = await restClient.getSubAccountId()
+  return new AddressBookClient(subAccountId, restClient)
+}
+
+export async function createAddressBookClient(
+  endpoint: string,
+  apiKey: string
+): Promise<AddressBookClient> {
+  const restClient = new RestClient(endpoint, apiKey)
+  return await createInstance(restClient)
+}
```

Comment:
You have to use an exported method to create a client when you then have to provide to the exported class? Why not just create the client as part of the class constructor?

Reply:
since I don't need clientId for AddressBookClient it could be simplified in this way. will do.
other clients (AgentClient, for example) need to fetch clientId via API, so that's why it's not done in the constructor.

Reply:
refactored.
now all magic happens in AddressBook Constructor, but I kept simplified createAddressBookClient() method since it could be expected by Tenfold, since other classes they use have such methods.